vllm/logprobs.py

```diff
@@ -186,21 +185,16 @@ def append_logprobs_for_next_position(
     # We do not need a special case for the sampled token
     # being in the topk, since inserting duplicated data
     # into a dictionary twice is the same as doing it once.
-    topk_ranks = range(1, num_logprobs + 1)
-    ranks = itertools.chain((rank,), topk_ranks)
+    ranks = [rank] + list(range(1, num_logprobs + 1))
```

Comment:
This is about 1.4x faster if we want to micro-optimise

```suggestion
    ranks = [rank, *range(1, num_logprobs + 1)]
```